app/models/mixins/aggregation_mixin/methods.rb

```diff
@@ -39,13 +39,10 @@ def all_storages
     end
 
     def aggregate_hardware(from, field, targets = nil)
-      from      = from.to_s.singularize
-      select    = field == :aggregate_cpu_speed ? "cpu_total_cores, cpu_speed" : field
-      targets ||= send("all_#{from}_ids")
-      targets   = targets.collect(&:id) unless targets.first.kind_of?(Integer)
-      hdws      = Hardware.where("#{from}_id" => targets).select(select)
-
-      hdws.inject(0) { |t, hdw| t + hdw.send(field).to_i }
+      from_singular = from.to_s.singularize
+      targets ||= send("all_#{from}")
+      targets   = targets.collect(&:id) if targets.kind_of?(Array) && !targets.first.kind_of?(Integer)
+      Hardware.where("#{from_singular}_id" => targets).sum(field)
```

Comment:
For reference, we probably could get rid of that `config/brakeman.ignore` entry by using this here:

http://api.rubyonrails.org/classes/ActiveRecord/Sanitization/ClassMethods.html#method-i-sanitize_sql_array

```ruby
where_clause = sanitize_sql_array(["#{from_singular}_id IN (?)", targets])
Hardware.where(where_clause).sum(field)
```

Might have to change the `if targets.kind_of?(Array) && !targets.first.kind_of?(Integer)` a bit more just to make sure you are dealing with an Array of ints, but should work.